service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/utils/AuthenticationUtils.java

```diff
@@ -43,10 +43,21 @@
 public class AuthenticationUtils {
```

Comment:
As a general rule, static Util classes do more harm than good. Please consider using injectable (and replaceable, through the use of Interfaces) collaborators.

Reply:
old code, I just removed SecureRandom instantiation inside the method since it's a costly operation

Reply:
We will have to discuss this, I understand it's debatable.
While I can appreciate that in theory PRs should be as laser focused as possible, I believe that works only in a semi-sane environment - which is not our case.
The only way to get out of the mess of code we have at the moment is to fix things as soon as you have to touch them for whatever reason - otherwise we are always postponing fixing issues to an hypothetical future "big cleanup" - which will never be scheduled due to external pressure, as history confirms.